### PR TITLE
Support getting cluster metrics in 2.0 with auth enabled

### PR DESCRIPTION
**Pull request by actgardner**
Switch metric collection to use the internal API, so we don't need an auth token to list pipelines and repos.

This also fixes a subtle bug in `ListRepo` which would have prevented users from listing repos if there was any repo they didn't have read access to. I added a test to make sure it won't regress, but this really highlights the need to pull all the auth checks in PFS into a common place - ideally high in the call stack so it's very clear what permission is required for each RPC.

### CODE REVIEW

File: src/server/auth/server/testing/auth_test.go [hunk 1]
```diff
@@ -1103,6 +1103,10 @@ func TestListAndInspectRepo(t *testing.T) {
 	require.Equal(t,
 		buildBindings(alice, auth.RepoOwnerRole), getRepoRoleBinding(t, aliceClient, repoNone))
 
+	// put a file in the repo Bob can't access - we need to be able to get the size of the commits
+	err := aliceClient.PutFile(client.NewCommit(repoNone, "master", ""), "/test", strings.NewReader("test"))
```
Comment (acohen4) [inline src/server/auth/server/testing/auth_test.go:1107]: should we assert that there's a certain amount of bytes in the repo?

**Review by acohen4 — APPROVED** [review]
Just one question on the test. Otherwise LGTM

File: src/server/auth/server/testing/auth_test.go (hunk 1 above)
Comment (actgardner) [reply src/server/auth/server/testing/auth_test.go:1107]: I'm not too concerned about asserting that here, PFS has it's own test that the results are accurate, here we're just testing the permissions model.

**Comment by chainlink** [comment]
Niice! cc @mindthevirt

**Comment by mindthevirt** [comment]
Love it! Thank you @actgardner

**Pull request merged**